spec/models/user_spec.rb

```diff
@@ -768,6 +768,17 @@
           UpdateUser.new(user: user, attributes: { unique_session_id: mock_session_id }).call
         end
 
+        it 'creates SuspendedEmail records for each email address' do
+          user_email_address = user.email_addresses.last
+          expect(SuspendedEmail).to receive(:generate_email_digest).with(user_email_address.email).
+            and_return('digest1')
+          expect(SuspendedEmail).to receive(:create!).with(
+            digested_base_email: 'digest1',
+            email_address: user_email_address,
+          )
+          user.suspend!
```

Comment:
I would have this check for the real side effect rather than stub:
```suggestion
          expect { user.suspend! }.to(change { SuspendedEmail.count }.by(1))
```